### PR TITLE
Fixes typo in Group::findGroupByPath found by sonarcloud

### PR DESCRIPTION
**Pull request by brad2014**
## Description
Hi.  A first small contribution. As I'm working my way through the codebase to learn it, I noticed an error pointed out by Sonar, and made a few adjustments which, to fix it and, I think, make the intention of the code a bit clearer (to me at least). If you think it's useful, I offer it.

Group.cpp had 
```
if (!groupPath.endsWith("/") && !groupPath.endsWith("/"))
```
which SonarCloud flagged (it clearly was not what was intended, though it would only in practice create an incorrect false positive if basePath were not normalized).

## How has this been tested?
Reran tests, which continue to all pass. The resulting refactor is meant to have the same functionality. 

## Types of changes

- ✅ Refactor (non-breaking changes which make code clearer,  more maintainable, or less risky)

## Checklist:
- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -12,3 +12,4 @@ release*/
 .DS_Store
 .version
 \.scannerwork/
+bw-output/
```
Comment (droidmonkey) [inline .gitignore]: Please revert this change

File: src/core/Group.cpp [hunk 2]
```diff
@@ -577,30 +577,36 @@ Entry* Group::findEntryByPath(QString entryPath, QString basePath)
...
+    QString normalizedGroupPath;
+
+    if (groupPath == "") {
+        normalizedGroupPath = QString("/"); // root group
+    } else {
+        normalizedGroupPath = ((groupPath.startsWith("/"))? "" : "/")
+            + groupPath
+            + ((groupPath.endsWith("/") )? "" : "/");
     }
+    return findGroupByPathRecursion(normalizedGroupPath, "/");
```
Comment (droidmonkey) [inline src/core/Group.cpp]: Why not just make the findGroupByPath function recursive?

Comment (brad2014) [reply src/core/Group.cpp]: My thinkin was that the public entrypoint findGroupByPath does input normalization, but once normalized, the internal recursive step doesn't need to redo that normalization.

Comment (droidmonkey) [reply src/core/Group.cpp]: Gotcha
